web-common/src/features/dashboards/url-state/filters/convertors.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  createAndExpression,
+  createBinaryExpression,
+  createInExpression,
+  createOrExpression,
+  createSubQueryExpression,
+  getAllIdentifiers,
+} from "@rilldata/web-common/features/dashboards/stores/filter-utils";
+import { V1Expression, V1Operation } from "@rilldata/web-common/runtime-client";
+import grammar from "./expression.js";
+import nearley from "nearley";
+
+export function convertFilterParamToExpression(filter: string) {
+  const parser = new nearley.Parser(nearley.Grammar.fromCompiled(grammar));
+  parser.feed(filter);
+  return convertToExpression(parser.results[0]);
+}
+
+const BinaryOperationMap = {
+  "=": V1Operation.OPERATION_EQ,
+  "!=": V1Operation.OPERATION_NEQ,
+  ">": V1Operation.OPERATION_GT,
+  ">=": V1Operation.OPERATION_GTE,
+  "<": V1Operation.OPERATION_LT,
+  "<=": V1Operation.OPERATION_LTE,
+};
+function convertToExpression(parsed: any) {
+  let subExpr: V1Expression | undefined;
+  switch (typeof parsed) {
+    case "object":
```

Comment:
Given there's only 1 case here, maybe a guard clause would be better?

```ts
if (typeof parsed !== "object") return parsed
```

Reply:
Removed